File: exercises/04_exercise_for_loops/08_tennis_ranklist.py
import math
num_tournaments = int(input())
initial_points = int(input())
total_points = initial_points
won_tournaments = 0
for tournaments in range (1, num_tournaments + 1):
    position = input()
    if position == "W":
        total_points += 2000
        won_tournaments += 1
    elif position == "F":
        total_points += 1200
    elif position == "SF":
        total_points += 720
average_points = (total_points - initial_points) / num_tournaments
percent_won_tournaments = won_tournaments / num_tournaments * 100
print(f"Final points: {math.floor(total_points)}")
print(f"Average points: {math.floor(average_points)}")
print(f"{percent_won_tournaments:.2f}%")

File: 05_JavaScriptBasics/02_Exercises/04_ForLoop/08_tennisRanklist.js
function tennisRanklist(input){
    let numTournaments = Number(input[0]);
    let initialPoints = Number(input[1]);
    let result = "";
    let wonTournaments = 0;
    let points = 0;

    for (let i=1; i <= numTournaments; i++){
        result = input[i+1];

        switch (result){
            case "W":
                points += 2000;
                wonTournaments += 1;
                break;
            case "F":
                points += 1200;
                break;
            case "SF":
                points += 720;
                break;
        }
    }

    let averagePoints = points / numTournaments;
    let percentWonTournaments = wonTournaments / numTournaments * 100;
    let totalPoints = initialPoints + points;

    console.log(`Final points: ${totalPoints}`);
    console.log(`Average points: ${Math.floor(averagePoints)}`);
    console.log(`${percentWonTournaments.toFixed(2)}%`)
}

tennisRanklist(["5", "1400", "F", "SF", "W", "W", "SF"]);
tennisRanklist(["4", "750", "SF", "W", "SF", "W"]);
tennisRanklist(["7", "1200", "SF", "F", "W", "F", "W", "SF", "W"]);
